modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/wstep/impl/CertificateEnrollmentServiceImpl.java

```diff
@@ -126,101 +118,54 @@ public void requestSecurityToken(String tokenType, String requestType,
                 headerBinarySecurityToken = element.getFirstChild().getNextSibling().getFirstChild().getTextContent();
             }
         }
-
+        List<ConfigurationEntry> tenantConfigurations;
+        try {
+            if (getTenantConfigurationData() != null) {
+                tenantConfigurations = getTenantConfigurationData();
+            } else {
+                String msg = "Tenant configurations are not initialized.";
+                log.error(msg);
+                throw new WindowsDeviceEnrolmentException(msg);
+            }
+        } catch (DeviceManagementException e) {
+            String msg = "Error occurred in while getting tenant configurations.";
```

Comment:
I think you can avoid unnecessary variable creation like above.. Please check other places as well..